### ISSUE TITLE
Poor evaluation results

### ISSUE THREAD

**Issue by Edric-star** [noahzn/Lite-Mono#172]
Hi, noahzn, thanks again for your support to my last post.
Since I started to reproduce your training process, I set all the parameters as default, and somehow the eval results were not nice though, could you take a look at my results? After reviewing almost all the previous posts that I had changed my pytorch version from newer one to 1.12.1, which didn't fix the poor eval res.
Below are one example of my evaluation res:

  
```
Mono evaluation - using median scaling
 Scaling ratios | med: 46.200 | std: 0.231

   abs_rel |   sq_rel |     rmse | rmse_log |       a1 |       a2 |       a3 | 
&   0.314  &   4.202  &   9.004  &   0.383  &   0.571  &   0.798  &   0.901  \\

  flops: 5.032G, params: 3.069M, flops_e: 4.314G, params_e:2.842M, flops_d:718.295M, params_d:226.627K
```

BTW, I have download your pretrained model in imageNet (lite-mono-pretrain.pth), did I miss something else that resulting in my poor results?

**Comment by noahzn** [comment]
Hi, can you show me your command to run the script?

**Comment by Edric-star** [comment]
`python evaluate_depth.py --load_weights_folder ./tmp/mytrain/models/weights_29 --data_path /mnt/sdb/kitti_data  --model lite-mono`

**Comment by Edric-star** [comment]
`python3 train.py --data_path /mnt/sdb/kitti_data --model_name mytrain --num_epochs 30 --batch_size 12 --num_workers 12 --save_frequency 4 --mypretrain ./pth/lite-mono-pretrain.pth  --lr 0.0001 5e-6 31 0.0001 1e-5 31`

**Comment by Edric-star** [comment]
I used part of the kitti data (but basically over 50% of the whole amount), I only used kitti data from 2011_09_26, and there are no other files under my kitti_data since I didn't download 2011_09_28, 2011_10_03 .....

**Comment by noahzn** [comment]
> I used part of the kitti data (but basically over 50% of the whole amount), I only used kitti data from 2011_09_26, and there are no other files under my kitti_data since I didn't download 2011_09_28, 2011_10_03 .....

What test set did you use then? I cannot give any suggestions if you are using different training/testing sets...

**Comment by Edric-star** [comment]
you mean the gt? I generated it by export_gt_depth.py

**Comment by Edric-star** [comment]
For val_files.txt I deleted these rows where I didn't use

> > I used part of the kitti data (but basically over 50% of the whole amount), I only used kitti data from 2011_09_26, and there are no other files under my kitti_data since I didn't download 2011_09_28, 2011_10_03 .....
> 
> What test set did you use then? I cannot give any suggestions if you are using different training/testing sets...

**Comment by noahzn** [comment]
can you show the training image visualization in tensorboard

**Comment by Edric-star** [comment]
![Image](https://github.com/user-attachments/assets/6ca81968-65e2-4c91-b0c4-fea7e392b5a8)

> can you show the training image visualization in tensorboard

**Comment by noahzn** [comment]
Could you show the predicted images as well? You can switch to the other label on tensorboard, not this loss page.

**Comment by Edric-star** [comment]
![Image](https://github.com/user-attachments/assets/525eb7db-501f-4e9f-87fe-f44afd1b0595)

> Could you show the predicted images as well? You can switch to the other label on tensorboard, not this loss page.

There you go, please check it

**Comment by noahzn** [comment]
The pictures look good to me. I think you need to train more epochs now that you are using fewer images. Enough data is very important for self-supervised training.

**Comment by Edric-star** [comment]
I understand I'll try to enlarge my datasets,  thank you so much for your support.



---- Replied Message ----
| From | ***@***.***> |
| Date | 02/24/2025 18:55 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [noahzn/Lite-Mono] Poor evaluation results (Issue #172) |

The pictures look good to me. I think you need to train more epochs now that you are using fewer images. Enough data is very important for self-supervised training.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

noahzn left a comment (noahzn/Lite-Mono#172)

The pictures look good to me. I think you need to train more epochs now that you are using fewer images. Enough data is very important for self-supervised training.
